.github/workflows/features.yml

```diff
@@ -34,15 +30,27 @@ jobs:
           profile: minimal
           toolchain: stable
 
-      - name: Run check
+      - name: Check benchmarks
+        uses: actions-rs/cargo@v1
+        env:
+          RUSTFLAGS: "-D warnings"
+        # NOTE: benchmarks are limited to compiling only on non windows systems, without any features enabled. 
+        if: ${{ matrix.os != 'windows-latest' && matrix.features == '--no-default-features' }}
+        with:
+          command: clippy
+          args: --verbose --all-targets -p benchmarks
+      
+      - name: Check rumqttc and rumqttd
         uses: actions-rs/cargo@v1
+        env:
+          RUSTFLAGS: "-D warnings"
         with:
-          # TODO: migrate from check to clippy when many hints are applied
-          command: check
-          args: --verbose --all-targets ${{ matrix.features }}
+          command: clippy
+          args: --verbose --all-targets ${{ matrix.features }} -p rumqttc -p rumqttd
 
+        # NOTE: Tests for rumqttc and rumqttd
       - name: Run tests
         uses: actions-rs/cargo@v1
         with:
           command: test
-          args: --verbose --release ${{ matrix.features }}
+          args: --verbose --release ${{ matrix.features }} -p rumqttc -p rumqttd
```

Comment:
Test with sub-crates being unspecified would fail on msvc targets as benchmarks uses pprof that doesn't support it. Also, there are no tests in benchmarks...